### PR TITLE
mongo-c: make openssl recommended

### PR DESCRIPTION
**Pull request by TylerBrock**

### CODE REVIEW

File: Library/Formula/mongo-c.rb [hunk 1]
```diff
@@ -14,11 +14,17 @@ class MongoC < Formula
...
   def install
     # --enable-sasl=no: https://jira.mongodb.org/browse/CDRIVER-447
-    system "./configure", "--prefix=#{prefix}", "--enable-sasl=no"
+    args = ["./configure", "--prefix=#{prefix}", "--enable-sasl=no"]
+
+    if build.with?('openssl')
+      args << "--enable-ssl=yes"
+    end
+
+    system args
```
Comment (jacknagel) [inline Library/Formula/mongo-c.rb]: system does not accept an array directly.

**Comment by jacknagel** [comment]
What is the rationale for this?

If I turn it off, will it not pick up the system openssl instead?

File: Library/Formula/mongo-c.rb [hunk 2]
```diff
@@ -14,11 +14,17 @@ class MongoC < Formula
 
   depends_on "pkg-config" => :build
   depends_on "libbson"
-  depends_on "openssl"
+  depends_on "openssl" => :recommended
 
   def install
     # --enable-sasl=no: https://jira.mongodb.org/browse/CDRIVER-447
-    system "./configure", "--prefix=#{prefix}", "--enable-sasl=no"
+    args = ["./configure", "--prefix=#{prefix}", "--enable-sasl=no"]
```
Comment (mikemcquaid) [inline Library/Formula/mongo-c.rb]: Make the args array without `./configure` in it; more standard with other formulae.

**Comment by TylerBrock** [comment]
I'm one of the developers working on the mongo-c-driver and we are in the process of adding native TLS/SSL support for Mac OS X (and possibly windows later) using secure transport. I could imagine a day very soon where openssl dependency is downgraded to simply "optional" as secure transport would be the preferred way to use TLS with the c driver on OSX.

Therefore "recommended" better describes the relationship between libmongoc and openssl right now. It's definitely recommended but not required, and may simply be optional in the future.

**Comment by mikemcquaid** [comment]
> If I turn it off, will it not pick up the system openssl instead?

That's the important question here with this option.

**Comment by TylerBrock** [comment]
Oh, good question. We can make it do either. What it should do depends on whether you take `--without-openssl` to mean "build this package without openssl at all" or "build this package without the homebrew ssl".

Which should it be?

**Comment by mikemcquaid** [comment]
The prior; the system SSL has been deprecated and should no longer be used for security reasons. Thanks!

**Comment by TylerBrock** [comment]
Ok, great, thanks for making that clear. I've updated the pull request so that if you say --without-openssl you explicitly disable it.

**Comment by mikemcquaid** [comment]
Great, thanks.

**Pull request closed without merging**